native/src/components/RemoteContent.tsx

```diff
@@ -57,29 +72,67 @@ const RemoteContent = (props: RemoteContentProps): ReactElement | null => {
     }
   }, [onLoad, webViewHeight, content])
 
-  // messages are triggered in renderHtml.ts
-  const onMessage = useCallback((event: WebViewMessageEvent) => {
-    const message = JSON.parse(event.nativeEvent.data)
-    if (message.type === HEIGHT_MESSAGE_TYPE && typeof message.height === 'number') {
-      setWebViewHeight(message.height)
-      return
-    }
+  useFocusEffect(
+    useCallback(() => {
+      appSettings.loadExternalSourcePermissions().then(setExternalSourcePermissions).catch(reportError)
+    }, []),
+  )
 
-    if (message.type === WARNING_MESSAGE_TYPE) {
-      log(message.message, 'warning')
-    } else {
-      const error = new Error(message.message ?? 'Unknown message received from webview')
-      reportError(error)
-      setError(error.message)
+  // Webview content does not reload on opt-in click
+  useEffect(() => {
+    if (reloadSettings) {
+      appSettings.loadExternalSourcePermissions().then(setExternalSourcePermissions).catch(reportError)
+      setReloadSettings(false)
     }
-  }, [])
+  }, [reloadSettings])
+
+  // messages are triggered in renderHtml.ts
+  const onMessage = useCallback(
+    (event: WebViewMessageEvent) => {
+      const message = JSON.parse(event.nativeEvent.data)
+      if (message.type === HEIGHT_MESSAGE_TYPE && typeof message.height === 'number') {
+        setWebViewHeight(message.height)
+        return
+      }
+
+      if (message.type === SETTINGS_MESSAGE_TYPE && typeof message.openSettings === 'boolean') {
+        navigateTo({ route: CONSENT_ROUTE })
+        return
+      }
+
+      if (message.type === IFRAME_MESSAGE_TYPE && typeof message.allowedSource === 'object') {
+        const updatedSources = updateSourcePermissions(externalSourcePermissions, message.allowedSource)
+        appSettings.setExternalSourcePermissions(updatedSources).catch(reportError)
+        setReloadSettings(true)
+        return
+      }
+
+      if (message.type === WARNING_MESSAGE_TYPE) {
+        log(message.message, 'warning')
+      } else {
+        const error = new Error(
+          message.message ? JSON.stringify(message.message) : 'Unknown message received from webview',
+        )
+        reportError(error)
+        setError(error.message)
+      }
+    },
+    [externalSourcePermissions, navigateTo],
+  )
 
   const onShouldStartLoadWithRequest = useCallback(
     (event: WebViewNavigation): boolean => {
+      if (buildConfig().allowedIframeSources.some(source => event.url.indexOf(source) > 0)) {
```

Comment:
```suggestion
      if (buildConfig().whitelistedIframeSources.some(source => event.url.indexOf(source) > 0)) {
```

Reply:
Why? I think we should not use the terms `whitelisted` or `blacklisted` anymore because of discriminating language.

Reply:
Jap. You're right. Just couldn't test this without changing 🤷🏼‍♀️